Classes/Domain/Repository/CreatorRepository.php

```diff
@@ -13,42 +13,156 @@
 
 namespace SUDHAUS7\Sudhaus7Wizard\Domain\Repository;
 
-use TYPO3\CMS\Extbase\Persistence\Generic\Query;
-use TYPO3\CMS\Extbase\Persistence\Repository;
+use Doctrine\DBAL\DBALException;
+use Doctrine\DBAL\Driver\Exception;
+use SUDHAUS7\Sudhaus7Wizard\Domain\Model\Creator;
+use TYPO3\CMS\Core\Database\Connection;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Database\Query\QueryBuilder;
+use TYPO3\CMS\Core\DataHandling\DataHandler;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
 
 /**
  * Class CreatorRepository
  */
-class CreatorRepository extends Repository
+class CreatorRepository
 {
-    public function findAll()
+    protected static string $table = 'tx_sudhaus7wizard_domain_model_creator';
+
+    protected QueryBuilder $queryBuilder;
+
+    public function __construct(
+        ConnectionPool $connection
+    ) {
+        $this->queryBuilder = $connection->getQueryBuilderForTable(self::$table);
```

Comment:
Don't store a QueryBuilder instance directly as property. QueryBuilder instances must not be reused, and per design changed on things. So either it needs to be ensured that this is not used directly but instead cloned to a variable before usage.

Better save the connection here and use $this->connection->createQueryBuilder()` in the methods instead.